src/osd/scrubber/osd_scrub_sched.cc

```diff
@@ -800,3 +800,22 @@ void ScrubQueue::dump_scrub_reservations(ceph::Formatter* f) const
   f->dump_int("scrubs_remote", scrubs_remote);
   f->dump_int("osd_max_scrubs", conf()->osd_max_scrubs);
 }
+
+void ScrubQueue::clear_pg_scrub_blocked(spg_t blocked_pg)
+{
+  dout(5) << fmt::format(": pg {} is unblocked", blocked_pg) << dendl;
+  --blocked_scrubs_cnt;
```

Comment:
We might increment the counter few times (for a single blocked scrub) and `clear_pg_scrub_blocked()` only once. Resulting in never resetting the count.

Reply:
Not sure I see how that might happen. If you are correct - this is obviously a bug.
I thought I got all scenarios covered?

Reply:
https://github.com/ceph/ceph/pull/46643#discussion_r899119719